### PR TITLE
Issue 2249: Add more logs to test case to troubleshoot end to end test case

### PR DESCRIPTION
**Pull request by arvindkandhare**
**Change log description**
This adds more logging to find out exact operation that is blocking the completion of end to end test cases.

**Purpose of the change**
To debug #2249.

**What the code does**
Adds logs after each of the future completion.

**How to verify it**
The logs are printed.

### CODE REVIEW

**Comment by andreipaduroiu** [comment]
It might be good to enable debug logging for this test alone and only run it as part of the build.

File: segmentstore/server/src/test/java/io/pravega/segmentstore/server/store/StreamSegmentStoreTestBase.java [hunk 1]
```diff
@@ -230,7 +232,10 @@ private ServiceBuilder createReadOnlyBuilder() throws Exception {
         for (String segmentName : segmentNames) {
             // Add half the attribute updates now.
             for (int i = 0; i < halfAttributeCount; i++) {
-                segmentFutures.add(store.updateAttributes(segmentName, createAttributeUpdates(), TIMEOUT));
+                int finalI = i;
+                segmentFutures.add(store.updateAttributes(segmentName, createAttributeUpdates(), TIMEOUT).thenRun(() -> {
+                    log.info("Completed attrs {}", finalI);
```
Comment (fpj) [inline segmentstore/server/src/test/java/io/pravega/segmentstore/server/store/StreamSegmentStoreTestBase.java]: This is debug level, not info.

File: segmentstore/server/src/test/java/io/pravega/segmentstore/server/store/StreamSegmentStoreTestBase.java [hunk 2]
```diff
@@ -230,7 +232,10 @@ private ServiceBuilder createReadOnlyBuilder() throws Exception {
         for (String segmentName : segmentNames) {
             // Add half the attribute updates now.
             for (int i = 0; i < halfAttributeCount; i++) {
-                segmentFutures.add(store.updateAttributes(segmentName, createAttributeUpdates(), TIMEOUT));
+                int finalI = i;
```
Comment (fpj) [inline segmentstore/server/src/test/java/io/pravega/segmentstore/server/store/StreamSegmentStoreTestBase.java]: Call it `iteration`.

Comment (tkaitchuck) [reply segmentstore/server/src/test/java/io/pravega/segmentstore/server/store/StreamSegmentStoreTestBase.java]: or perhaps "completedAttrs"

File: segmentstore/server/src/test/java/io/pravega/segmentstore/server/store/StreamSegmentStoreTestBase.java [hunk 3]
```diff
@@ -239,12 +244,18 @@ private ServiceBuilder createReadOnlyBuilder() throws Exception {
                 lengths.put(segmentName, lengths.getOrDefault(segmentName, 0L) + appendData.length);
                 recordAppend(segmentName, appendData, segmentContents);
 
-                segmentFutures.add(store.append(segmentName, appendData, createAttributeUpdates(), TIMEOUT));
+                int finalI = i;
```
Comment (tkaitchuck) [inline segmentstore/server/src/test/java/io/pravega/segmentstore/server/store/StreamSegmentStoreTestBase.java]: same here

Comment (arvindkandhare) [reply segmentstore/server/src/test/java/io/pravega/segmentstore/server/store/StreamSegmentStoreTestBase.java]: Fixed.

File: segmentstore/server/src/test/java/io/pravega/segmentstore/server/store/StreamSegmentStoreTestBase.java (hunk 2 above)
Comment (arvindkandhare) [reply segmentstore/server/src/test/java/io/pravega/segmentstore/server/store/StreamSegmentStoreTestBase.java]: Fixed.

File: segmentstore/server/src/test/java/io/pravega/segmentstore/server/store/StreamSegmentStoreTestBase.java (hunk 1 above)
Comment (arvindkandhare) [reply segmentstore/server/src/test/java/io/pravega/segmentstore/server/store/StreamSegmentStoreTestBase.java]: Fixed.

**Comment by arvindkandhare** [comment]
Closed as underlying issues were cornered and fixed one by one.

**Pull request closed without merging**